pkg/migrate/migrate.go

```diff
@@ -485,6 +538,50 @@ func mutatePV(ctx context.Context, w *log.Logger, clientset k8sclient.Interface,
 	}
 }
 
+// get a SC, apply the selected mutator to the SC, update the SC, use the supplied validator to wait for the update to show up
+func mutateSC(ctx context.Context, w *log.Logger, clientset k8sclient.Interface, scName string, mutator func(sc *storagev1.StorageClass) (*storagev1.StorageClass, error), checker func(sc *storagev1.StorageClass) bool) error {
+	tries := 0
+	for {
+		sc, err := clientset.StorageV1().StorageClasses().Get(ctx, scName, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get storage classes %s: %w", scName, err)
+		}
+
+		sc, err = mutator(sc)
+		if err != nil {
+			return fmt.Errorf("error creating mutated SC: %w", err)
+		}
+
+		_, err = clientset.StorageV1().StorageClasses().Update(ctx, sc, metav1.UpdateOptions{})
+		if err != nil {
+			if k8serrors.IsConflict(err) {
+				if tries > 5 {
+					return fmt.Errorf("failed to mutate SC %s: %w", scName, err)
+				}
+				w.Printf("Got conflict updating SC %s, waiting 5s to retry\n", scName)
+				time.Sleep(time.Second * 5)
+				tries++
+				continue
+			} else {
+				return fmt.Errorf("failed to mutate SC %s: %w", scName, err)
+			}
+		}
+		break
+	}
+
+	for {
+		sc, err := clientset.StorageV1().StorageClasses().Get(ctx, scName, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get storage classes %s: %w", scName, err)
+		}
+
+		if checker(sc) {
+			return nil
+		}
+		time.Sleep(time.Second * 5)
```

Comment:
is this supposed to try forever?

Reply:
If this takes more than a minute, something has gone badly wrong - but I don't know if that's worth handling

Reply:
makes sense